### DISCUSSION TITLE
extra step to compile, run, build and test changes

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#440] [Q&A] [answered]
Congratulations from the newly released version 12, @orbitalquark (and each contributor).

I read at https://orbitalquark.github.io/textadept/manual.html#compile-run-build-and-test-changes:

> All compile, run, build, and test commands no longer fire immediately when invoked. Instead, candidate commands are displayed in the command entry first. Pressing `Enter` will run the command. This allows for in-place modifications of commands that will be remembered next time the command is run for a particular file/project.

Sorry, but is there any way to get the previous behavior (not to require extra user input)?

I would be surprised, if I would be the only one that this new feature (which doesn’t seem to be mentioned in https://orbitalquark.github.io/textadept/changelog.html) causes only issues (besides the extra input, forgetting to press the key and thinking that something is wrong with compilation). I hope it can be at least deactivated.

Many thanks for your help.

**Comment by orbitalquark** [comment]
It took me a while to get used to it too. I think the only way to go back to the old behavior is to edit *modules/textadept/run.lua* from this: https://github.com/orbitalquark/textadept/blob/f08b6c3314072b3e68be95ee0fc78dd05ac8ec26/modules/textadept/run.lua#L149-L150

to

```
command_entry_f[id](command, working_dir or dir, env, event, commands, key, macros)
```

Sorry for the trouble :(

**Reply by ousia** [reply]
Sorry, not marking it as an answer yet, @orbitalquark (just in case a better solution can be found).

As far as I understand, running (compiling or whatever) without extra interaction has been the default previous to current version of Textadept.

Sorry, it is not easy for me (as an average user) to understand why the default behavior has been modified (explaining the rationale may help). I cannot see the gain.

But even if the default has been changed, I cannot understand why the previous default was removed (as far as I have experienced, not requiring extra input after firing compilation has been the default in all editors I have used).

Would it be too much to ask to provide a configurable parameter (in the spirit of the one proposed by @merlindiavova) to disable the extra confirmation?

Many thanks for your help.

**Comment by merlindiavova** [comment]
Could we not enable the old behavior via a flag?

i.e `textadept.run.inspect_before_exec = false`

Otherwise, those who want that behavior will have update that file after every update

**Comment by orbitalquark** [comment, marked as answer]
The rationale behind the confirmation is that I often want to change arguments to run/compile/build/test commands, and I'd like to keep a history of those changes (like in a command line). Think `Makefile` rules, names of tests to run, arguments to shell scripts, etc.

An option to bypass this sounds reasonable.

As for why this option is not present is because when I announced this change back in October 2022, nobody spoke up.

**Reply by ousia** [reply]
Many thanks for considering an option to bypass the new default, @orbitalquark.

**Reply by orbitalquark** [reply]
Thanks for asking so nicely :)

**Reply by ousia** [reply]
As I first experienced the issue, I thought it would be a problem for me.

But it would be extremely stupid of me to think there was no reason at all for this new feature (no matter it woudn‘t fit everybody).

Many thanks for your excellent work.

**Comment by orbitalquark** [comment]
I've added a new `textadept.run.run_without_prompt` option: https://github.com/orbitalquark/textadept/commit/a74562ccdd55ae94cb7f2a4d5e9f45a11e62de41

It will be in the next nightly build.

**Reply by ousia** [reply]
Many thanks for the new commit, @orbitalquark.

I use TA both at work and at home. If I‘m the only user affected by this, please ignore this comment.

At work, no matter what the code is, only software builds are allowed. Sorry if I’m not describing this right, but it seems a sensible policy to me.

I moved back to TA-11.4 (so the extra step would not be an issue for me), but it might be worth considering whether a new version 12.1 could be released containing this and other minor improvements.

Again, if I’m the only user affected by this, I‘m fine with version 11.4 (and this comment could be completely ignored).

Many thanks for your help.

**Reply by orbitalquark** [reply]
If you are allowed to use 12.0, then you can manually edit *modules/textadept/run.lua*, making the change in the commit above (copy-paste text). Hopefully that would be acceptable? No executable code was changed. I don't think 12.1 will be released in September. Probably October.

**Reply by ousia** [reply]
That would be fine (or so I guess 😅).

Many thanks for the idea to fix this.
